### PR TITLE
HCMPRE-400 : plan config name - implement fuzzy search

### PR DESCRIPTION
**Pull request by tanishi-egov**

### CODE REVIEW

**Review by Priyanka-eGov — DISMISSED** [review]

File: health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java [hunk 1]
```diff
@@ -105,8 +105,8 @@ private String buildPlanConfigSearchQuery(PlanConfigurationSearchCriteria criter
 
         if (criteria.getName() != null) {
             addClauseIfRequired(preparedStmtList, builder);
-            builder.append(" pc.name = ?");
-            preparedStmtList.add(criteria.getName());
+            builder.append(" pc.name LIKE ?");
+            preparedStmtList.add(criteria.getName() + '%');
```
Comment (shashwat-egov) [inline health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java:109]: move % string literal to constants file

**Review by shashwat-egov — CHANGES_REQUESTED** [review]

File: health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java (hunk 1 above)
Comment (tanishi-egov) [reply health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java:109]: done

**Review by shashwat-egov — APPROVED** [review]

**Comment by shashwat-egov** [comment]
There are conflicts in this branch, please resolve

**Comment by tanishi-egov** [comment]
yes, done.

**Review by Priyanka-eGov — APPROVED** [review]

**Review by shashwat-egov — APPROVED** [review]

File: health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import org.springframework.stereotype.Component;
 import org.springframework.util.CollectionUtils;
 import org.springframework.util.ObjectUtils;
+import static digit.config.ServiceConstants.*;
```
Comment (Priyanka-eGov) [inline health-services/plan-service/src/main/java/digit/repository/querybuilder/PlanConfigQueryBuilder.java:12]: no need to import all the ServiceConstants, just import your constant

**Review by Priyanka-eGov — CHANGES_REQUESTED** [review]

**Review by Priyanka-eGov — APPROVED** [review]

**Pull request merged**